src/actions/MastodonAddKernelsAction.C

```diff
@@ -0,0 +1,135 @@
+    {
+      // Velocity AuxKernels: These Auxkernels are added as MooseObjectActions
+      // due to conflicts with DynamicTensorMechanicsAction when added using
+      // addAuxKernel method. TODO: Investigate this conflict. Talk to Ben Spencer.
+      InputParameters action_params = _action_factory.getValidParams("AddKernelAction");
+      // Create the action
+      action_params.set<std::string>("type") = "NewmarkVelAux";
+      std::shared_ptr<MooseObjectAction> vel_action = std::static_pointer_cast<MooseObjectAction>(
+          _action_factory.create("AddKernelAction", vel_auxkernel[j], action_params));
+      vel_action->appendTask("add_aux_kernel");
+      // Assigning input parameters
+      InputParameters & vel_kernel_params = vel_action->getObjectParams();
+      vel_kernel_params.set<AuxVariableName>("variable") = vel_variables[j];
+      vel_kernel_params.set<std::vector<VariableName>>("acceleration") = {accel_variables[j]};
+      vel_kernel_params.set<Real>("gamma") = gamma;
+      vel_kernel_params.set<ExecFlagEnum>("execute_on") = EXEC_TIMESTEP_END;
+      // Add the action to the warehouse
+      _awh.addActionBlock(vel_action);
+
+      // Acceleration AuxKernels: Added using the addAuxKernel method
+      InputParameters accel_kernel_params = _factory.getValidParams("NewmarkAccelAux");
+      accel_kernel_params.set<AuxVariableName>("variable") = accel_variables[j];
+      accel_kernel_params.set<std::vector<VariableName>>("displacement") = {disp_variables[j]};
+      accel_kernel_params.set<std::vector<VariableName>>("velocity") = {vel_variables[j]};
+      accel_kernel_params.set<Real>("beta") = beta;
+      accel_kernel_params.set<ExecFlagEnum>("execute_on") = EXEC_TIMESTEP_END;
+      _problem->addAuxKernel("NewmarkAccelAux", accel_auxkernel[j], accel_kernel_params);
+    }
+
+    // stress and strain AuxKernels: Added using the addAuxKernel method.
+    const std::vector<std::string> stress_names = {
+        "stress_xx", "stress_yy", "stress_xy", "stress_zz", "stress_zx", "stress_yz"};
+    const std::vector<std::string> strain_names = {
+        "strain_xx", "strain_yy", "strain_xy", "strain_zz", "strain_zx", "strain_yz"};
+    const std::vector<unsigned int> index_i = {0, 1, 1, 2, 2, 1};
+    const std::vector<unsigned int> index_j = {0, 1, 0, 2, 0, 2};
+    const std::vector<std::size_t> matrix_size = {1, 3, 6}; // =1 for 1D, =3 for 2D, and =6 for 3D
+    for (std::size_t k = 0; k < matrix_size[_problem->mesh().dimension() - 1]; k++)
+    {
+      InputParameters stress_kernel_params = _factory.getValidParams("RankTwoAux");
+      stress_kernel_params.set<MaterialPropertyName>("rank_two_tensor") = "stress";
+      stress_kernel_params.set<AuxVariableName>("variable") = stress_names[k];
+      stress_kernel_params.set<unsigned int>("index_i") = index_i[k];
+      stress_kernel_params.set<unsigned int>("index_j") = index_j[k];
+      _problem->addAuxKernel("RankTwoAux", stress_names[k], stress_kernel_params);
```

Comment:
The stress and strain auxkernels also need to be evaluated only at end of time step.

Reply:
I will change this.